### PR TITLE
[FEATURE] New Asana Sources and Improvements

### PR DESCRIPTION
**Pull request by gsomoza**
### Added
New Asana sources:
- New Task Assigned in Project
- Updated Task in Project Filtered by Given Fields

New features for existing sources:
- **task-updated-in-project** can now emit events only for tasks assigned to a given user GID

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

**Comment by gsomoza** [comment]
@GTFalcao closed my other PRs and opened this one for all features

File: components/asana/sources/task-updated-in-project/task-updated-in-project.mjs
Comment (GTFalcao) [inline components/asana/sources/task-updated-in-project/task-updated-in-project.mjs:10]:
```suggestion
  version: "1.1.0",
```

**Review by GTFalcao — COMMENTED** [review]
Hey @gsomoza excellent work grouping the PRs! 🚀 

I left a couple of comments about version numbers and also one change that was not brought from the individual PRs. This is almost good to go!

Also something I forgot and the automated checks have pointed out - since the `asana.app` file was updated and other existing components import it, they all need to have their patch version incremented too. (this is not optimal since the changes you made are irrelevant to these other components, but it's a "safety catch" to ensure none of them miss out on updates should the changes be relevant)

**Comment by gsomoza** [comment]
@GTFalcao applied the changes you requested. FYI it looks like PR checks don't run until a maintainer approves, so I don't know if my changes will pass them. Is there a way to run them in the Gitpod environment?

**Comment by GTFalcao** [comment]
Hey @gsomoza nice work! This looks good to go! I'm moving it forward for QA.

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Asana_5648_1043.pdf](https://github.com/PipedreamHQ/pipedream/files/10938256/Asana_5648_1043.pdf)

**Comment by dannyroosevelt** [comment]
@gsomoza let us know if you want to debug and make the appropriate changes or if you want someone from our team to finish it up

**Comment by gsomoza** [comment]
@dannyroosevelt I really don't mind if you guys jump in to help! I'm in UTC+2 so I wouldn't be able to get back to this until next week anyway.

**Comment by dannyroosevelt** [comment]
> @dannyroosevelt I really don't mind if you guys jump in to help! I'm in UTC+2 so I wouldn't be able to get back to this until next week anyway.

No worries at all -- @GTFalcao do you want to wrap this up?

**Comment by GTFalcao** [comment]
Yeah @dannyroosevelt sorry I missed this notification - I can check this out and finish up whatever's left!

**Review by luancazarine — APPROVED** [review]

**Pull request merged**
